enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+---
+
+
+# Automated Backups of etcd
+
+## Summary
+
+Enable the automated backups of etcd snapshots and other metadata necessary to restore an openshift cluster from a quorum loss scenario.
+
+## Motivation
+
+The [current documented procedure](https://docs.openshift.com/container-platform/4.12/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html) for performing a backup of an OpenShift
+cluster is manually initiated. This reduces the likelihood of a timely backup being available during a disaster recovery scenario.
+
+The procedure also requires gaining a root shell on a control plane
+node. Shell access to OpenShift control plane nodes access is generally
+discouraged due to the potential for affecting the reliability of the
+node.
+
+### Goals
+
+- One-time cluster backup can be triggered without requiring a root shell
+- Scheduled backups can be configured after cluster installation
+- Backups are saved locally on the host filesystem of control-plane nodes
+- This feature is validated with an e2e restore test that ensures the backups saved can be used to recover the cluster from a quorum loss scenario
+
+
+### Non-Goals
+- Have automated backups enabled by default with cluster installation
+- Save cluster backups to PersistentVolumes or cloud storage
+  - This would be a future enhancement or extension to the API
+- Automate cluster restoration
+
+
+### User Stories
+
+- As a cluster administrator I want to initiate a cluster backup without requiring a root shell on a control plane node so as to minimize the risk involved
+- As a cluster administrator I want to schedule recurring cluster backups so that I have a recent cluster state to recover from in the event of quorum loss (i.e. losing a majority of control-plane nodes)
+- As a cluster administrator I want to have failure to take cluster backups for more than a configurable period to be reported to me via critical alerts
+
+
+## Proposal
+
+### Workflow Description
+
+To enable automated backups a new singleton CRD can be used to specify the backup schedule and other configuration as well as monitor the status of the backup operations. 
+
+A new controller in the cluster-etcd-operator would then reconcile this CRD to save backups locally per the specified configuration.
+
+### API Extensions
+
+A new CRD `EtcdBackup` will be introduced to the API group `config.openshift.io` with the version `v1alpha1`. The CRD would be feature-gated until the prerequisite e2e test has an acceptable pass rate. 
+
+The spec and status structures will be as listed below:
+
+
+```Go
+// EtcdBackupSpec represents the configuration of the EtcdBackup CRD.
+type EtcdBackupSpec struct {
+
+    // Reason defines the reason for the most recent backup.
+    // Setting this to a value different from the most recent reason will trigger a one-time backup
+    // The default "" means no backup is triggered
+    // +kubebuilder:default:=""
+    // +optional
+    Reason string `json:"reason"`
```

Comment:
In our existing operators, we have a `forceRedeploymentReason` or some such, right?  Given your need to track backups, it may be more practical to have an object like the Insights collection that indicates a desire to run a backup "now".